### DISCUSSION TITLE
Does anybody serve as a VAR for Serenity hosting?

### DISCUSSION THREAD

**Discussion by LSNicholls** [serenity-is/Serenity#6672] [General]
Hi folks,

I am working with a consulting company to develop a Serenity/Start Sharp practice.

While they have deep experience with databases, they need to provide an equally polished front end occasionally, and I believe that this framework fits the bill for them.  I am still in the exploration phase, but it's looking pretty good so far.

But they will not want to get into the business of being hosting providers or managing Azure instances for their clients.

So, I am interested in any and all experiences that you may have had with companies who would be responsible for deployment and maintenance of Azure or AWS environments and fill this niche for them.

Or, if you want to say "Lisa, that's ridiculous,Serenity is basically an xcopy deploy, why does anybody need a VAR for hosting? The client provides the environment, and it is up to them to manage it, get the certificate, etc"  Maybe that is true.

But ... Consider the multi-tenant scenario that is worked out in the Serenity tutorials.  In at least one case, the consulting company is going to want to provide a platform using a multi-tenant setup, rather than just hand off to a client.  Do any of you do this?  Do you have dedicated personnel to handle the management of the environment, etc?

I was thinking that, while we're never going to get to WordPress provider sophistication (https://learn.microsoft.com/en-us/azure/app-service/quickstart-wordpress) there might be somebody in this group who would provide these services to other developers.

Thanks for any and all thoughts.

**Comment by reach2rv** [comment]
Hi @LSNicholls ,

I have done something sort of this by making changes to Linux hosting panel and deploying multiple apps on single Linux instance using the same. 

The panel handles everything getting SSL from let's encrypt/ deployment, multiple domains, support for docker and database management etc.

Are you looking out for something like this?

**Comment by LSNicholls** [comment]
Thank you so much for writing, @reach2rv. This might work for them.   Are you in the business of managing this type of thing, serving as a VAR?

 By "multiple apps", do you mean multiple instances of a Serenity app, multiple different Serenity apps, management of multiple tenants, or what?

As I understand it, their desire for multi-tenancy is for ease of application updates, but I'm not actually sure it is the right path for them, if they do go with Serenity for this business case.  If you have an organized pipeline for pushing updates, potentially to multiple instances, it shouldn't make all that much difference.  Multi-tenancy has other advantages for analysis, that's for sure.  Just thinking out loud.

**Reply by reach2rv** [reply]
Hey @LSNicholls ,

I  am not a VAR, but do have a AWS instance to serve my clients with serenity hosting as most of them are not from tech background or don't have IT team in place. ( I charge them for per year) 

Multiple apps means multiple serenity instances. I have never worked on multi tenancy with Serenity, as my each client have different requirements though they are from one industry. 

My setup is each app instance with its own database

I have 2 Linux servers 1 acting as app server where I have serenity app deployment for each client and other one for database to which each client connect using internal AWS ip.

For auto update, it is not possible with my setup as in my case I have a single codebase for all clients and I do configuration on deployment.

Hope this helps.

**Comment by LSNicholls** [comment]
It does help a lot @reach2rv .  Your setup and approach is what I would think is good for Serenity as well, and kind of what I expected.  

Even with a multi-tenancy approach, I am pretty sure this consulting firm is going to have to deal with setting up an AWS or Azure instance, etc.  I do not believe they will find a "Serenity provider" like a "WordPress hosting specialist" would be, but wanted to check and see if I had missed something.  

Thank you so much.

**Reply by reach2rv** [reply]
If opportunity persists, I would love to work on such project. Please do let me know if I can be at any help on this. Thank you!

**Comment by LSNicholls** [comment]
Thank you for letting me know, @reach2rv /Arvind!
